api/sip/client.py

```diff
@@ -705,14 +707,22 @@ def read_message(self, max_size=1024*1024):
         data = ""
         tmp = ""
         while not done:
-            tmp = self.socket.recv(4096)
+            try:
+                tmp = self.socket.recv(4096)
+            except socket.timeout:
+                raise socket.error
```

Comment:
socket.error here is a class, not an exception object. Since socket.timeout is a subclass of socket.error, the simplest thing to do would be to take these two lines out and let the next two lines handle the problem. You could also raise a more specific IOError than is raised in the next two lines.

But, I expect you're reraising the exception as `IOError` so it can be dealt with by `except (IOError, socket.error), e:` on line 361. I don't think that's necessary either, since `socket.error` and `socket.timeout` (a subclass of `socket.error`) are already covered.

Looking at `circulation/tests/sip/test_client.py:TestNetworkError.retry_on_initial_timeout` I also see that we have a test that verifies we create a new socket connection when a timeout happens.

Reply:
This is exactly the kind of help I needed. Thanks. It's what happens when a neophyte gets in the middle of a codebase :-). Yes, I was looking at getting the handler starting line 361 involved, since I was unsure how errors trickled up the code stack, and was mocking a block I saw in articles elsewhere. A couple suggested handling socket.timeout errors in an except block prior to general exception in order to differentiate. Perhaps that would better be handled, if desired, in the 361 block.